src/service_daemon.rs

```diff
@@ -770,6 +770,18 @@ struct IntfSock {
     sock: Socket,
 }
 
+impl IntfSock {
+    /// Returns the interface index and the associated ip version
+    fn intf_idx_ip_ver(&self) -> (u32, u8) {
+        let ip_ver = match self.intf.addr {
+            IfAddr::V4(_) => 4u8,
+            IfAddr::V6(_) => 6u8,
+        };
+
+        (self.intf.index.unwrap_or(0), ip_ver)
```

Comment:
We're using fallback 0 in `set_scope_id`, but in terms of actual network interfaces, it seems their index [starts from 1](https://man7.org/linux/man-pages/man3/if_nameindex.3.html).   We probably want to check the returned index is not 0 to be treated as valid.

Reply:
Yes makes sense, I'll add the checks.